synapse/storage/databases/main/client_ips.py

```diff
@@ -29,8 +41,32 @@
 LAST_SEEN_GRANULARITY = 120 * 1000
 
 
+LastClientIpByDeviceEntry = TypedDict(
```

Comment:
Ooooh `TypedDict`! ooi, was there a reason you preferred the function-call form over the dataclass-like form? (i.e. a class that inherits from TypedDict)

Reply:
I didn't know the dataclass-like form existed. Which style do we prefer?

Reply:
Quick grep suggests dataclass-like, e.g.

https://github.com/matrix-org/synapse/blob/94b620a5edd6b5bc55c8aad6e00a11cc6bf210fa/synapse/handlers/register.py#L81-L85

Reply:
Let's go with that then. Do you have any suggestions for better names for these `TypedDict`s?

Reply:
I think they're okay as they are. I'm guessing that `Entry` means that they correspond directly to a database table (or some subset of its columns?)

Reply:
I've converted them to class-forms and tried to give them better names. Hopefully they make sense.

At the time `Entry` really didn't mean anything, though I guess they do correspond to columns in database tables. That's an implementation detail internal to the data store, so perhaps it's not the best name.